src/components/pill-selector-dropdown/AvatarPill.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+import React from 'react';
+import noop from 'lodash/noop';
+import classNames from 'classnames';
+import X from '../../icon/fill/X16';
+// $FlowFixMe this imports from a typescript file
+import LabelPill from '../label-pill';
+import Avatar from '../avatar';
+
+import './AvatarPill.scss';
+
+type Props = {
+    avatarUrl?: string,
+    className?: string,
+    hasWarning?: boolean,
+    id?: string | number,
+    isDisabled?: boolean,
+    isExternal?: boolean,
+    isSelected?: boolean,
+    isValid?: boolean,
+    onRemove: () => any,
+    showAvatar?: boolean,
+    text: string,
+};
+
+const RemoveButton = ({ onClick, ...rest }: { onClick: () => any }) => (
+    <X {...rest} aria-hidden="true" onClick={onClick} />
+);
+
+const AvatarPill = ({
+    isDisabled = false,
+    isSelected = false,
+    hasWarning = false,
+    isExternal,
+    isValid = true,
+    onRemove,
+    text,
+    className,
+    showAvatar = false,
+    avatarUrl,
+    id,
+}: Props) => {
+    const styles = classNames('bdl-AvatarPill', className, {
+        'bdl-AvatarPill-isSelected': isSelected && !isDisabled,
+        'bdl-AvatarPill-isDisabled': isDisabled,
+    });
+
+    let pillType;
+
+    if (hasWarning) {
+        pillType = 'warning';
+    }
+
+    if (!isValid) {
+        pillType = 'error';
+    }
+
+    const handleClickRemove = isDisabled ? noop : onRemove;
+
+    const avatar = showAvatar ? (
+        <Avatar id={id} isExternal={isExternal} name={text} size="small" avatarUrl={avatarUrl} shouldShowExternal />
```

Comment:
What happens if you use `LabelPill.Icon` here? ideally, whatever we put inside `LabelPill.Pill` will be recognized sub-components of label pill only.

Reply:
good callout, it does work